build/deploy-test-resources.yml

```diff
@@ -0,0 +1,83 @@
+name: Arcus Observability - Deploy test resources
+
+trigger: none
+pr: none
+
+parameters:
+  - name: azureServiceConnection
+    displayName: 'Azure service connection'
+    type: string
+    default: 'Azure Codit-Arcus Service Principal'
+  - name: resourceGroupName
+    displayName: 'Resource group name'
+    default: arcus-observability-dev-we-rg
+  - name: appInsightsName
+    displayName: 'Application Insights name'
+    default: arcus-observability-dev-we-app-insights
+  - name: variableGroupName
+    displayName: 'Variable group name'
+    default: Arcus Observability - Integration Testing
+
+variables:
+  - name: keyVaultName
+    value: 'arcus-observability-kv'
+  - template: ./variables/build.yml
+
+stages:
+  - stage: Deploy
```

Comment:
Best that we also add the self-contained integration tests as a second depend stage, so that we always have a verification that the deployment went successful. Since they are not expensive, we can use them as some kind of smoke tests.